lib/resources/backup.js

```diff
@@ -115,12 +116,12 @@ const backup = (keys, response) => {
   // get our configuration and obtain a tmpdir to dump the database into.
   // we don't bother catching any errors here since we can't return any kind of
   // useful message to the user anyway, and there is nothing to clean up.
-  const dump = (tmpdir) => pgdump(tmpdir.path).then(() => {
+  tmp.withDir((tmpdir) => pgdump(tmpdir.path).then(() => {
     // halt the keepalive.
     release();
 
     // TODO: mostly copypasta for now from lib/task/fs:
-    promisify(readdir)(tmpdir.path).then((files) => {
+    return promisify(readdir)(tmpdir.path).then((files) => {
```

Comment:
Did this line make a big difference? The rest of it makes sense. This change seems like a good idea but now I don't get what it was doing before.

Reply:
i think it was not working before lol